<launch>
  <arg name="bag_file" doc="path to the bagfile whose messages should be reordered" />
  <arg name="ref_topic" default="!" doc="topic name which acts as a reference for the time stamps" />
  <arg name="out_path" default="!" doc="path to file with reordered bag file" />
  <arg name="is_recording" default="false" doc="only set true when running this launch file together with a recording node. Makes the data_processing wait until the recording is finished" />

  <arg name="ref_resolved" value="$(eval '' if arg('ref_topic')=='!' else '--ref_topic ' + arg('ref_topic'))" />
  <arg name="out_resolved" value="$(eval '' if arg('out_path')=='!' else '--out_path ' + arg('out_path'))" />

  <node name="fixBag3D" pkg="roboy_ad"
        type="fixBag3D.py"
        args="$(arg bag_file) $(arg ref_resolved) $(arg out_resolved) --is_recording $(arg is_recording)" 
        output="screen"/>

</launch>
